app/models/generic_event.rb

```diff
@@ -52,4 +60,12 @@ def self.from_event(event)
 
     type.constantize.from_event(event)
   end
+
+  def self.inherited(subclass)
```

Comment:
We want to make sure that all STI classes are included in the static STI_CLASSES list. Alternative to this is to work out the constants by filename/grepping the constants that the GenericEvent model defines. The second approach isn't great because it breaks lazy loading in dev mode and we have to anticipate what to grep for. The first approach might break if a developer decides to put something else in the same generic events directory. This approach gives the richest feedback and is easier to test.